docker/Dockerfile

```diff
@@ -0,0 +1,90 @@
+FROM ubuntu:22.04
+LABEL maintainer="Marco Randazzo"
+
+# Non-interactive installation mode
+ENV DEBIAN_FRONTEND=noninteractive
+ENV DOCKER_UPDATED_ON=23_11_2023
+
+# Install essentials
+RUN apt-get update && apt-get install -y apt-utils software-properties-common sudo psmisc lsb-release protobuf-compiler libatlas-base-dev \
+    tmux nano geany vim wget curl build-essential git gitk cmake cmake-curses-gui autoconf xserver-xorg-video-dummy xserver-xorg-legacy \
+    net-tools terminator libjpeg-dev ffmpeg apt-transport-https ca-certificates gnupg libace-dev ycm-cmake-modules locales \
+    python3-setuptools python3-pip iproute2 python3-tornado swig lsof iftop iputils-ping gdb bash-completion btop mlocate && rm -rf /var/lib/apt/lists/*
+
+RUN pip3 install numpy bpytop
+
+RUN sed -i 's/allowed_users=console/allowed_users=anybody/' /etc/X11/Xwrapper.config
+COPY ./common/xorg.conf /etc/X11/xorg.conf
+
+# Install yarp dependencies
+RUN echo "deb http://www.icub.org/ubuntu `lsb_release -cs` contrib/science" > /etc/apt/sources.list.d/icub.list
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 57A5ACB6110576A6
+RUN apt-get update && apt-get install -y icub-common libopencv-dev && sudo rm -rf /var/lib/apt/lists/*
```

Comment:
The http://www.icub.org/ubuntu repository is not updated and will be deactivated on 1st of January 2024 (https://github.com/orgs/robotology/discussions/653) so probably you can directly install dependencies of yarp instead of using `icub-common`.

Reply:
The dependencies installed by icub-common are the one in https://github.com/robotology/icub-main/blob/v2.4.0/.ci/packages_vars.sh#L26 .

Reply:
fixed thnx!

Reply:
@elandini84 tour-guide-robot docker images need to be updated as well